### DISCUSSION TITLE
Several ideas

### DISCUSSION THREAD

**Discussion by xsteve** [TatriX/tshell#3] [Ideas]
I like the concept of this shell mode.

Some things that I have observed when I tested the mode

a) tshell-help has an extra closing paren - the code can't be loaded

b) A key binding to change the shell prompt would be nice

c) Support for other languages (e.g. python) could also be very interesting.
   I guess some kind of prolog is needed then (to predefine functions or to import modules)

**Comment by xsteve** [comment]
d) Adding a license would also be useful (I would like to package emacs-tshell for guix ;-))

**Reply by TatriX** [reply]
Thanks!
a) Fixed
b) Do we really need it? When I want to run specific interpreter I usually just `C-j >` or `C-j $` or something similar.
c) Could be interesting in the long run, though it seems it's more suitable for org-babel
d) Done :)

**Comment by xsteve** [comment]
One thing that is missing is a "(require 'transient)" to ensure that transient is loaded

b) the prompt is not self explanatory. Perhaps "elisp> ", "sh> ", "bash> ", "zsh> " would be better for users that use tshell from time to time
c) I guess you are correct

An interesting use case for me could be a kind of literate programming sheet with a bunch of commands that I want to replay from time to time.

**Reply by TatriX** [reply]
I've made prompt customizeable though limited to a simple string.

> An interesting use case for me could be a kind of literate programming sheet with a bunch of commands that I want to replay from time to time.

That's how I use it right now I guess. For example I have something like this in one of my projects:
```sh
$ # Run server
$ <*server*> cargo run
$ <*server*> ls # ^ this will run process in a buffer named "*server*"

$ # Install `xh`
$ sudo pacman -S xh
$ xh --help

$ # Test stuff
> (setenv "url" "localhost:8000/api")
$ echo $url
$ xh $url/pingz
$ xh $url/stuff | jq .
$ xh post $url/stuff x-foo:bar
$ echo 'select * from users' | psql

```

**Reply by xsteve** [reply]
Thanks for your fast response.
I have sent my guix package definitions to the guix-patches list now ;)
